message-bridge/src/main/java/org/acme/message/bridge/MessageBridgeRoute.java

```diff
@@ -0,0 +1,46 @@
+package org.acme.message.bridge;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import org.apache.camel.builder.RouteBuilder;
+
+@ApplicationScoped
+public class MessageBridgeRoute extends RouteBuilder {
+
+    @Override
+    public void configure() throws Exception {
+        restConfiguration().component("servlet");
+        rest()
+                .post("/message")
+                .id("rest")
+                .to("direct:publish");
+
+        from("direct:publish")
+                .id("ibmmq")
+                .transacted()
+                .log("Sending message to IBMMQ: ${body}")
+                .to("ibmmq:queue:{{ibm.mq.queue}}?disableReplyTo=true");
+
+        from("ibmmq:queue:{{ibm.mq.queue}}")
+                .id("ibmmq-amq")
+                .transacted()
+                .choice()
+                .when(simple("${header.JMSRedelivered}"))
+                .log("Redelivering message after rollback to ActiveMQ: ${body}")
+                .otherwise()
+                .log("Sending message from IBMMQ to ActiveMQ: ${body}")
+                .end()
+                .to("amq:queue:{{amq.queue}}")
+                .process(ex -> {
+                    if (ex.getIn().getBody(String.class).toLowerCase().contains("rollback")) {
+                        if (!ex.getIn().getHeader("JMSRedelivered", Boolean.class)) {
```

Comment:
Why we check `JMSRedelivered` here?

Reply:
The idea might be to rollback the transaction only once. So that we see a rollback followed by successful commit. Maybe a comment could help readability.

Reply:
yes, @aldettinger is correct, just some way how to proceed with the transaction after the rollback, I'll add a comment there